core/object/message_queue.h

```diff
@@ -115,13 +135,22 @@ class MessageQueue {
 	Error push_notification(Object *p_object, int p_notification);
 	Error push_set(Object *p_object, const StringName &p_prop, const Variant &p_value);
 
+	Error flush();
+	void clear();
 	void statistics();
-	void flush();
 
 	bool is_flushing() const;
-
 	int get_max_buffer_usage() const;
 
+	CallQueue(Allocator *p_custom_allocator = 0, uint32_t p_max_pages = 8192, const String &p_error_text = String());
+	virtual ~CallQueue();
+};
+
+class MessageQueue : public CallQueue {
```

Comment:
That makes sense, in fact, it could be renamed to MainThreadMessageQueue, but it requires touching a lot of other code for the PR, so I think its probably better to leave this for a separate PR renaming it.